modules/settings/settings.tsx

```diff
@@ -1,20 +1,143 @@
 import React, { useState } from "react";
 import { useTranslations } from "next-intl";
-import SideBar from "@/core/components/sidebar";
+import { useForm } from "react-hook-form";
 import { getStaticPropsWithTranslations } from "@/modules/lang/props";
 import { GlobalCore } from "@/core/module/module.types";
 import settings_styles from "./styles/settings.module.css";
+import messageHandler from "@/core/message-handler";
+import AuthService from "@/services/auth/auth-supabase.service";
+import { settings_schema } from "@/core/forms/schemas/auth-schema";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { FaTrash } from "react-icons/fa6";
+import { GetStaticProps } from "next";
 
-export const getStaticProps = getStaticPropsWithTranslations;
+export const getStaticProps: GetStaticProps = getStaticPropsWithTranslations;
 
 const Settings = () => {
   const t = useTranslations("");
+  const [language, setLanguage] = useState("en");
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(settings_schema),
+  });
+
+  const onSubmit = async (data: any) => {
+    const updatedUser = await AuthService.updateUser(data.email, data.password);
+
+    if (updatedUser)
+      messageHandler.handleSuccess("Profile updated successfully");
+  };
+
+  const handleDeleteAccount = () => {
+    console.log("Delete account requested");
```

Comment:
Do we need a ticket for this? (You can create tickets :) and put them in the backlog)